tap/tlstapper/tls_poller.go

```diff
@@ -131,16 +131,39 @@ func (p *tlsPoller) handleTlsChunk(chunk *tlsChunk, extension *api.Extension,
 		p.readers[key] = reader
 	}
 
-	reader.timer.CaptureTime = time.Now()
-	reader.chunks <- chunk
-
+	p.updateTlsReader(chunk, reader, ip, port)
+	
 	if os.Getenv("MIZU_VERBOSE_TLS_TAPPER") == "true" {
-		p.logTls(chunk, ip, port)
+		p.logTls(chunk, key, reader)
 	}
 
 	return nil
 }
 
+func (p *tlsPoller) updateTlsReader(chunk *tlsChunk, reader *tlsReader, ip net.IP, port uint16) {
+	myIp, myPort, err := getAddressBySockfd(p.procfs, chunk.Pid, chunk.Fd, chunk.isClient())
+
+	if err == nil {
+		// Currently tls key is built from the foreign address, pid, thread id and more, it doesn't contains 
+		//	the local address though. The reason is pure technical, and require traversing kernel structs in
+		//	the ebpf (hopefully we'll get to fix it).
+		//
+		// If there are many tcp requests to the same remote address, at around the same time, from different
+		//	connections, they all get the same tls key.
+		//
+		// If we don't update tcpID with the new local address, the requests matcher fails, because it have
+		//	the old tcp id used by the previous connection.
+		//
+		// Yet getAddressBySockfd may fail because the file descriptor already gone, usually for localhost 
+		// 	where everything is CPU, or for the last chunk in a big message. If that happen, its better to 
+		//	stay with the old tcpID which most chances is still valid, instead of replacing it with unknown.
+		//
+		reader.tcpID = p.buildTcpId(chunk, ip, port, myIp, myPort)
+	}
```

Comment:
This is almost the same as `buildTcpInitialId`.
Combine into 1 function with an option.